### ISSUE TITLE
Provide PL/I compile options for code points as Settings

### ISSUE THREAD

**Issue by kkrein** [IBM/zopeneditor-about#17]
Are there any plans to include additional settings for PL1?
For example the following statement results in an error:
   dcl v3 char(1024);
   v3 = 'abc' !! 'abc'; 
-> COMMENT_START  expected after this token
because currently only || is defined as concat operator. 

Also the default margin is set to 72, but we are having programs with margins up to 100.
It would be nice if we could customize such settings (like the PropertyGroups in RDz/IDz) or is there already a possibility which i am missing?
Best regards
kkrein

**Comment by phaumer** [comment]
@kkrein thanks. I keep this open for us to discuss Settings options for the future. 

Right now, you can use process directives in each file to declare the redefinition. See here: https://www.ibm.com/support/knowledgecenter/SSY2V3_5.2.0/com.ibm.ent.pl1.zos.doc/pg/optns.html and here https://www.ibm.com/support/knowledgecenter/SSY2V3_5.2.0/com.ibm.ent.pl1.zos.doc/lr/ctlproc.html

So for `||` you would do the same as `|`. Add the line `*process OR('!');` to your program and then the syntax errors should go away.

**Comment by phaumer** [comment]
For changing the margin you can define user or workspace specific settings just for PL/I. Add this to your User or Workspace settings:

```json
"[pl1]": {
        "editor.rulers": [
            0,
            1,
            100
        ]
    }
```

**Comment by kkrein** [comment]
Thanks for the quick reply, i'm aware of the process directives but they are not allowed in our company so this no option for us. So i'm hoping this enhancement is included in one of the next releases.

**Comment by phaumer** [comment]
Support `MARGINS(1,120)` and not only `*process MARGINS(2,120);`

**Comment by phaumer** [comment]
Fixed in Z Open Editor 1.1. We extended property groups with language and compiler option properties:
https://ibm.github.io/zopeneditor-about/Docs/setting_propertygroup.html#language-scoping
